pubtools/_content_gateway/push_base.py

```diff
@@ -0,0 +1,463 @@
+        version_id = self.pv_mapping.get((product_name, product_code, version_name))
+        all_files = self.cgw_client.get_files(product_id, version_id)
+        for data in all_files:
+            yield (product_name, product_code, version_name, data['downloadURL']), data.get('id')
+
+    def __init__(self, cgw_hostname, cgw_username, cgw_password):
+        """
+        Initialize.
+
+        Args:
+            cgw_hostname (str):
+                CGW registry URL
+            cgw_username (str):
+                username for CGW HTTP API
+            cgw_password (str):
+                password for CGW HTTP API
+        """
+
+        self.auth = CGWBasicAuth(cgw_username, cgw_password)
+        self.cgw_client = CGWClient(cgw_hostname, self.auth)
+        self.product_mapping = FetcherDict({}, fetcher=self._fetch_product, key_checker=self._product_mapping_key_check)
+        self.pv_mapping = FetcherDict({}, fetcher=self._fetch_product_version,
+                                      key_checker=self._product_version_mapping_key_check)
+        self.file_mapping = FetcherDict({}, fetcher=self._fetch_file, key_checker=self._file_key_check)
+
+    def process_product(self, item):
+        """
+        Responsible for product workflow operations such as Create,
+        Update and delete the product based on the satisfied condition.
+
+        The product operation follows the following workflow:
+            - Search for the product record from the product mapping
+            - If `product state` == `present` and record not found in mapping
+
+                - Creates a new product record on content gateway with given data
+                - Add the new record into the product mapping records
+
+            - If `product state` == `present` and record found in mapping
+
+                - Update the product record on content gateway with given data
+                - Update the record into the product mapping records
+
+            - If `product state` == `absent` and record found in mapping
+
+                - Delete the product record on the content gateway
+                - Remove the record from the product mapping records
+
+            - If `product state` == `absent` and record not found in mapping
+
+                - Raises CGWError error
+
+        Args:
+            item (list(dict)):
+                List of product record
+
+        Raises:
+            CGWError:
+                When product record is not present and tries to delete the same
+
+        Returns:
+            product_id (int):
+                Returns the product ID
+        """
+
+        product_name = item.get('metadata').get('name')
+        product_code = item.get('metadata').get('productCode')
+        LOG.debug(
+            "Fetching for product_id of product_name:- %s and product_code:- %s" % (product_name, product_code))
+
+        product_id = self.product_mapping.get((product_name, product_code))
+
+        if item.get('state') == 'present':
```

Comment:
I noticed the inconsistent quotes and other style errors. Does this repo have CI enabled? Things like black and flake8?

Reply:
flake8 is added in tox. And I don't know how it works and even for what purpose it is used for.